source/agora/consensus/data/Block.d

```diff
@@ -410,8 +411,9 @@ public Block makeNewBlock (Transactions)(const ref Block prev_block,
 
     block.header.merkle_root = block.buildMerkleTree();
     block.header.enrollments = enrollments;
+    block.header.enrollments.sort!((a, b) => a.utxo_key < b.utxo_key);
```

Comment:
Ah maybe I can tell `sort` it should be strictly monotonic. Let me read the docs.

Reply:
It doesn't seem to have this feature.

Reply:
Looking back on my old memories, I know that the enrollments are missing here because they are sorted from the DB.
If it's added here, isn't it possible to sort it twice?

at present here
https://github.com/bpfkorea/agora/blob/e72c0fbd60013b02303a705ce3d3718372081440/source/agora/consensus/EnrollmentPool.d#L266-L267

In the future
https://github.com/bpfkorea/agora/blob/e72c0fbd60013b02303a705ce3d3718372081440/source/agora/consensus/ValidatorSet.d#L259

Reply:
Is it because of the data change in the candidate process?

Reply:
It reduces the chance of bugs creeping in.

There's a much bigger cost in implicitly sorting the transactions. There will be just a few enrollments and only periodically and not in every block, but there will be many transactions in every block.